crowns.lic

```diff
@@ -2,84 +2,128 @@
   Documentation: https://elanthipedia.play.net/Lich_script_repository#dice-game
 =end
 
-arg_definitions = [
-  [
-    { name: 'bet', regex: /\d+/, optional: true, description: 'Optional override for betting amount.' }
-  ]
-]
-
-custom_require.call(%w[events common-travel common])
-
-settings = get_settings
-junk_list = settings.darkbox_junk.map { |x| /#{x}/i }
-dice_bet_amount = settings.dice_bet_amount || args.bet
-
-dice = ["anchor", "crown", "heart", "lightning", "ship", "trident"]
-
-Flags.add('bet', /"Place your bets!  Place your bets!"/)
-Flags.add('no_more_bets', "The libik yells")
-
-DRCT.walk_to(16306) unless (DRRoom.title.include?("Crown and Anchor, Table"))
-
-loop do
-  break if (DRRoom.title.include?("Crown and Anchor, Table"))
-  case DRC.bput("go cerulean curtain", "Reflecting", "As you peer")
-  when "Reflecting"
-    break
-  when "As you peer"
-    case DRC.bput("go carmine curtain", "Reflecting", "As you peer")
-    when "Reflecting"
-      break
-    when "As you peer"
-      pause 30
+custom_require.call(%w[events common-travel common common-items common-money])
+
+class Dice
+  include DRC
+  include DRCI
+  include DRCM
+  include DRCT
+
+  def initialize
+    arg_definitions = [
+    [
+      { name: 'bet', regex: /\d+/, optional: true, description: 'Optional override for betting amount.' }
+    ]
+    ]
+
+    settings = get_settings
+    #@junk_list = settings.darkbox_junk.map { |x| /#{x}/i }
+    @trash_items = settings.darkbox_junk
+    @dice_bet_amount = settings.dice_bet_amount || args.bet
+    @dice_money_on_hand = settings.dice_money_on_hand
+    @withdraw = settings.dice_withdraw
+
+    @dice = ["anchor", "crown", "heart", "lightning", "ship", "trident"]
+
+    Flags.add('bet', /"Place your bets!  Place your bets!"/)
+    Flags.add('no_more_bets', "The libik yells")
+
+    dice_game
+  end
+
+  def dice_game
+    loop do
+      check_money
+      walk_to_table
+      play_dice
+      check_prize
     end
   end
-end
 
-loop do
-  count = 0
-  dice.each do |id|
-    case DRC.bput("bet #{dice_bet_amount} on #{id}", /You bet/, /You can't bet right now./, /You don't have that many kronars!/)
-    when /You don't have that many kronars!/
-      DRC.message("******* Out of money! ********")
+  def check_money
+    get_money if check_wealth('kronars') < 200 && @withdraw
+  end
+
+  def get_money
+    DRC.message("*** Heading to bank to money! ***")
+    DRCT.walk_to(16_315)
+    case DRC.bput("withdraw #{@dice_money_on_hand} platinum", 'we are not lending money at this time', 'The clerk counts out')
```

Comment:
❓ Doesn't this result in something like `withdraw 10 platinum platinum`?

Reply:
Yes, every time you dip below 200 coppers.

Reply:
Does that command work with the denomination in there twice? I'm just surprised, is all

Reply:
Let me test that particular piece. It might not. I copied over from another script and modified it.

Reply:
Removed platinum from the line as it is in base. I like it in base because the person knows what it is grabbing.